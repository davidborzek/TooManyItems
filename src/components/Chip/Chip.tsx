import { StyleSheet, Text, TouchableOpacity } from 'react-native';

import { Ionicons } from '@expo/vector-icons';
import React from 'react';

type Props = {
  title: string;
  icon?: React.ComponentProps<typeof Ionicons>['name'];
  selected?: boolean;
  onPress?: () => void;
};

/**
 * Ein wiederverwendbarer Component welcher Chips bereitstellt.
 * https://m3.material.io/components/chips
 */
export default function Chip({ icon, title, selected, onPress }: Props) {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[styles.badge, selected ? styles.active : {}]}
    >
      {icon && <Ionicons style={styles.icon} name={icon} size={14} />}
      <Text>{title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  active: {
    backgroundColor: 'rgb(200, 150, 150)',
  },
  badge: {
    alignItems: 'center',
    backgroundColor: '#c1c1c1',
    borderRadius: 90,
    flexDirection: 'row',
    marginHorizontal: 5,
    padding: 10,
  },
  icon: {
    marginRight: 5,
  },
});
