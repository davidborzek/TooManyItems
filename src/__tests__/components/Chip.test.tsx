import Chip from '../../components/Chip/Chip';
import React from 'react';
import renderer from 'react-test-renderer';

it('Chip renders correctly with required props', () => {
  const tree = renderer.create(<Chip title="Some Title" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Chip renders correctly in selected state', () => {
  const tree = renderer.create(<Chip title="Some Title" selected />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Chip renders correctly with icon', () => {
  // Note: The icon causes a console.error only in test
  const tree = renderer.create(<Chip title="Some Title" icon="add" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Chip renders correctly with onPress', () => {
  const tree = renderer
    .create(
      <Chip title="Some Title" onPress={() => console.log('do something')} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
