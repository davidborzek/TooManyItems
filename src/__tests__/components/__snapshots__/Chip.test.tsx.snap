// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders correctly in selected state 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(200, 150, 150)",
      "borderRadius": 90,
      "flexDirection": "row",
      "marginHorizontal": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text>
    Some Title
  </Text>
</View>
`;

exports[`Chip renders correctly with icon 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#c1c1c1",
      "borderRadius": 90,
      "flexDirection": "row",
      "marginHorizontal": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text />
  <Text>
    Some Title
  </Text>
</View>
`;

exports[`Chip renders correctly with onPress 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#c1c1c1",
      "borderRadius": 90,
      "flexDirection": "row",
      "marginHorizontal": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text>
    Some Title
  </Text>
</View>
`;

exports[`Chip renders correctly with required props 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#c1c1c1",
      "borderRadius": 90,
      "flexDirection": "row",
      "marginHorizontal": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text>
    Some Title
  </Text>
</View>
`;
